src/lua/widget/container.c

```diff
@@ -154,16 +154,38 @@ static int container_numindex(lua_State*L)
       GtkWidget *searched_widget = g_list_nth_data(children,index);
       gtk_container_remove(GTK_CONTAINER(container->widget),searched_widget);
     } else {
-      luaL_error(L,"Incorrect index or value when setting the child of a container : you can only append, not change a child in the middle");
+      GList *current = g_list_nth(children, index);
+      gtk_container_remove(GTK_CONTAINER(container->widget), current->data);
+      for(current = current->next; current != NULL; current = current->next)
+      {
+        g_object_ref(current->data);
+        gtk_container_remove(GTK_CONTAINER(container->widget), current->data);
+      }
+
+      lua_widget widget;
+      luaA_to(L, lua_widget, &widget, 3);
+      gtk_container_add(GTK_CONTAINER(container->widget), widget->widget);
+
+      for(current = g_list_nth(children, index + 1); current != NULL; current = current->next)
+      {
+        gtk_container_add(GTK_CONTAINER(container->widget), current->data);
+        g_object_unref(current->data);
+      }
     }
     g_list_free(children);
     return 0;
   } else {
-    GtkWidget *searched_widget = g_list_nth_data(children,index);
+    if(index < 0 || index >= g_list_length(children))
+    {
+      lua_pushnil(L);
```

Comment:
and the same answer :) 

Even more, the old code for container doesn't raise an error, so behaviour of `container` and `combobox` was a bit inconsistent.